### PR TITLE
Exposing SSL-only version of MySQL Source

### LINKED ISSUE #6420: Create a secure-only MySQL source
## Tell us about the problem you're trying to solve
We want to create secure-only versions of connectors that can be used in the Airbyte cloud. The idea is that these connectors inherently prevent certain insecure connections such as connecting to a database over the public internet without encryption.

## Describe the solution you’d like
The acceptance criteria for this ticket is: 

- [ ] Create a new connector `source-<name>-secure` based on the current connector
- [ ] modify the connector's spec to hide any options which allow the user to disable TLS
- [ ] change the connector to enable TLS by default if the TLS option is not specified
- [ ] update connector documentation
- [ ] Encrypted connections are tested as part of either a custom integration test or acceptance test. Where possible, test it using a test container. If that's impossible and it must be tested on a real DB instance, create a DB instance in AWS ideally using terraform (but if TF is too hard just create it manually and make a ticket to encode it in TF)
- [ ] Create a PR 

## Implementation hints
[This PR](https://github.com/airbytehq/airbyte/pull/6362) contains an example of modifying a DB to only allow using TLS connections.

**Note**: to perform this task for any given connector, it must first support TLS as described in #5628

### PR DESCRIPTION
**Pull request by andriikorotkov**
## What
We want to provide versions of some connectors such that they only work in a "secure" mode. In the context of database connectors this means that we want the connector to only operate with TLS encryption turned on. While for the base version of the MySQL connector allows a user to send data unencrypted for users with a strong security posture and for our cloud product we want to enforce that data can never be sent across the public internet unencrypted.

## How
I created a new MySQL connector that just depends on the base MySQL connnector.

## Recommended reading order
1. MySqlSourceStrict.java

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR /airbyte
+
+ENV APPLICATION source-mysql-strict-encrypt
+
+COPY build/distributions/${APPLICATION}*.tar ${APPLICATION}.tar
+
+RUN tar xf ${APPLICATION}.tar --strip-components=1
+
+LABEL io.airbyte.version=0.1.0
+LABEL io.airbyte.name=airbyte/source-mysql-strict-encrypt
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/source-mysql-strict-encrypt/Dockerfile:12]: please, add a new line at the end of file

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/build.gradle [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    testImplementation testFixtures(project(':airbyte-integrations:connectors:source-jdbc'))
+    testImplementation project(':airbyte-test-utils')
+
+    testImplementation 'org.testcontainers:mysql:1.15.1'
+
+    integrationTestJavaImplementation project(':airbyte-integrations:bases:standard-source-test')
+
+    implementation files(project(':airbyte-integrations:bases:base-java').airbyteDocker.outputs)
+    integrationTestJavaImplementation files(project(':airbyte-integrations:bases:base-java').airbyteDocker.outputs)
+}
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/source-mysql-strict-encrypt/build.gradle:29]: the same about new line

**Review by etsybaev — APPROVED** [review]
LGTM. But please update jdbc connection string in new tests (setupEnvironment method) as discussed

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/Dockerfile (hunk 1 above)
Comment (andriikorotkov) [reply airbyte-integrations/connectors/source-mysql-strict-encrypt/Dockerfile:12]: updated it

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/build.gradle (hunk 2 above)
Comment (andriikorotkov) [reply airbyte-integrations/connectors/source-mysql-strict-encrypt/build.gradle:29]: updated it

**Review by alexandr-shegeda — APPROVED** [review]

**Review by etsybaev — APPROVED** [review]

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/MySqlSSLJdbcSourceAcceptanceTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+package io.airbyte.integrations.source.mysql;
+
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.string.Strings;
+import io.airbyte.db.Databases;
+import org.jooq.SQLDialect;
+import org.junit.jupiter.api.BeforeEach;
+
+class MySqlSSLJdbcSourceAcceptanceTest extends MySqlJdbcSourceAcceptanceTest {
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/MySqlSSLJdbcSourceAcceptanceTest.java:14]: Nitpick. `SSL` -> `Ssl`

```suggestion
class MySqlSslJdbcSourceAcceptanceTest extends MySqlJdbcSourceAcceptanceTest {
```

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.base.IntegrationRunner;
+import io.airbyte.integrations.base.Source;
+import io.airbyte.integrations.base.spec_modification.SpecModifyingSource;
+import io.airbyte.integrations.source.mysql.MySqlSource;
+import io.airbyte.protocol.models.ConnectorSpecification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySqlStrictEncryptSource extends SpecModifyingSource implements Source {
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java:17]: Can you add a javadoc to explain what the expected behavior is for this source and its difference from the `MySqlSource`?

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  private static final Logger LOGGER = LoggerFactory.getLogger(MySqlStrictEncryptSource.class);
+
+  MySqlStrictEncryptSource() {
+    super(MySqlSource.sshWrappedSource());
+  }
+
+  @Override
+  public ConnectorSpecification modifySpec(final ConnectorSpecification originalSpec) {
+    final ConnectorSpecification spec = Jsons.clone(originalSpec);
+    ((ObjectNode) spec.getConnectionSpecification().get("properties")).remove("ssl");
```
Comment (tuliren) [inline airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java:28]: Nitpick. Can you add a comment about why this property is removed?

File: airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/MySqlSSLJdbcSourceAcceptanceTest.java (hunk 3 above)
Comment (andriikorotkov) [reply airbyte-integrations/connectors/source-mysql/src/test/java/io/airbyte/integrations/source/mysql/MySqlSSLJdbcSourceAcceptanceTest.java:14]: Thanks for pointing that out. changed it

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java (hunk 5 above)
Comment (andriikorotkov) [reply airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java:28]: new comment was added

File: airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java (hunk 4 above)
Comment (andriikorotkov) [reply airbyte-integrations/connectors/source-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/source/mysql_strict_encrypt/MySqlStrictEncryptSource.java:22]: added javadoc

**Review by tuliren — APPROVED** [review]

**Pull request merged**